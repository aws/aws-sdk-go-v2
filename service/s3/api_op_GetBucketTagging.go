// Code generated by smithy-go-codegen DO NOT EDIT.

package s3

import (
	"context"
	awsmiddleware "github.com/aws/aws-sdk-go-v2/aws/middleware"
	"github.com/aws/aws-sdk-go-v2/aws/retry"
	"github.com/aws/aws-sdk-go-v2/aws/signer/v4"
	"github.com/aws/aws-sdk-go-v2/service/s3/types"
	smithy "github.com/awslabs/smithy-go"
	"github.com/awslabs/smithy-go/middleware"
	smithyhttp "github.com/awslabs/smithy-go/transport/http"
)

// Returns the tag set associated with the bucket. To use this operation, you must
// have permission to perform the s3:GetBucketTagging action. By default, the
// bucket owner has this permission and can grant this permission to others.  <p>
// <code>GetBucketTagging</code> has the following special error:</p> <ul> <li>
// <p>Error code: <code>NoSuchTagSetError</code> </p> <ul> <li> <p>Description:
// There is no tag set associated with the bucket.</p> </li> </ul> </li> </ul>
// <p>The following operations are related to <code>GetBucketTagging</code>:</p>
// <ul> <li> <p> <a>PutBucketTagging</a> </p> </li> <li> <p>
// <a>DeleteBucketTagging</a> </p> </li> </ul>
func (c *Client) GetBucketTagging(ctx context.Context, params *GetBucketTaggingInput, optFns ...func(*Options)) (*GetBucketTaggingOutput, error) {
	stack := middleware.NewStack("GetBucketTagging", smithyhttp.NewStackRequest)
	options := c.options.Copy()
	for _, fn := range optFns {
		fn(&options)
	}
	addawsRestxml_serdeOpGetBucketTaggingMiddlewares(stack)
	awsmiddleware.AddRequestInvocationIDMiddleware(stack)
	smithyhttp.AddContentLengthMiddleware(stack)
	AddResolveEndpointMiddleware(stack, options)
	v4.AddComputePayloadSHA256Middleware(stack)
	retry.AddRetryMiddlewares(stack, options)
	addHTTPSignerV4Middleware(stack, options)
	awsmiddleware.AddAttemptClockSkewMiddleware(stack)
	addClientUserAgent(stack)
	smithyhttp.AddErrorCloseResponseBodyMiddleware(stack)
	smithyhttp.AddCloseResponseBodyMiddleware(stack)
	addOpGetBucketTaggingValidationMiddleware(stack)
	stack.Initialize.Add(newServiceMetadataMiddleware_opGetBucketTagging(options.Region), middleware.Before)
	addUpdateEndpointMiddleware(stack, options)

	for _, fn := range options.APIOptions {
		if err := fn(stack); err != nil {
			return nil, err
		}
	}
	handler := middleware.DecorateHandler(smithyhttp.NewClientHandler(options.HTTPClient), stack)
	result, metadata, err := handler.Handle(ctx, params)
	if err != nil {
		return nil, &smithy.OperationError{
			ServiceID:     ServiceID,
			OperationName: "GetBucketTagging",
			Err:           err,
		}
	}
	out := result.(*GetBucketTaggingOutput)
	out.ResultMetadata = metadata
	return out, nil
}

type GetBucketTaggingInput struct {
	// The name of the bucket for which to get the tagging information.
	Bucket *string
}

type GetBucketTaggingOutput struct {
	// Contains the tag set.
	TagSet []*types.Tag

	// Metadata pertaining to the operation's result.
	ResultMetadata middleware.Metadata
}

func addawsRestxml_serdeOpGetBucketTaggingMiddlewares(stack *middleware.Stack) {
	stack.Serialize.Add(&awsRestxml_serializeOpGetBucketTagging{}, middleware.After)
	stack.Deserialize.Add(&awsRestxml_deserializeOpGetBucketTagging{}, middleware.After)
}

func newServiceMetadataMiddleware_opGetBucketTagging(region string) awsmiddleware.RegisterServiceMetadata {
	return awsmiddleware.RegisterServiceMetadata{
		Region:        region,
		ServiceID:     ServiceID,
		SigningName:   "s3",
		OperationName: "GetBucketTagging",
	}
}
