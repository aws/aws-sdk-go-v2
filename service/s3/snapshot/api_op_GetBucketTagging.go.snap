GetBucketTagging
	Initialize stack step
		RegisterServiceMetadata
		legacyEndpointContextSetter
		S3Shared:ARNLookup
		SetLogger
		OperationInputValidation
	Serialize stack step
		putBucketContext
		setOperationInput
		ResolveEndpoint
		S3:ProcessARNResource
		S3:ObjectLambdaEndpoint
		serializeImmutableHostnameBucket
		OperationSerializer
		S3:UpdateEndpoint
		S3:RemoveBucketFromPathMiddleware
	Build stack step
		ClientRequestID
		ComputeContentLength
		UserAgent
		RecursionDetection
	Finalize stack step
		DisableAcceptEncodingGzip
		ResolveAuthScheme
		GetIdentity
		ResolveEndpointV2
		disableHTTPS
		ComputePayloadHash
		SigV4ContentSHA256Header
		Retry
		RetryMetricsHeader
		setLegacyContextSigningOptions
		Signing
	Deserialize stack step
		AddRawResponseToMetadata
		ErrorCloseResponseBody
		CloseResponseBody
		ResponseErrorWrapper
		S3MetadataRetriever
		OperationDeserializer
		RecordResponseTiming
		RequestResponseLogger
